pkg/knowledge_base2/properties/resource_property.go

```diff
@@ -108,6 +108,11 @@ func (r *ResourceProperty) ZeroValue() any {
 }
 
 func (r *ResourceProperty) Contains(value any, contains any) bool {
+	if val, ok := value.(construct.ResourceId); ok {
+		if cont, ok := contains.(construct.ResourceId); ok {
+			return val == cont
```

Comment:
Should `Contains` be a match if `value` is a selector?
Eg. `Subnet.VPC.Contains("aws:vpc") == true`

Reply:
yeah it maybe should, the weird part is in the way its used today were more so checking for equals, but in the list,set,map case only care that a single index is equal which is why contains is called. Im wondering if we need a different method to cover this then